src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -265,7 +273,7 @@ public VersionStamp GetLatestProjectVersion()
     /// True if the solution contains a project with the specified project ID.
     /// </summary>
     public bool ContainsProject([NotNullWhen(returnValue: true)] ProjectId? projectId)
-        => projectId != null && ProjectStates.ContainsKey(projectId);
+        => projectId != null && GetProjectState(projectId) != null;
```

Comment:
```suggestion
        => GetProjectState(projectId) != null;
```

i think this works as i think GetProjectState is null safe.  if i'm wrong.  don't change this.

Reply:
I don't want to change this. The nullability constraints on GetProjectState indicate it shouldn't take in null, and with the new internal CompareTo method impl, that doesn't check for null.